### PR TITLE
build: Add `make nuke`; speed-up `make start`

### PR DESCRIPTION
**Pull request by alexec**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -444,7 +460,7 @@ $(GOPATH)/bin/goreman:
 
 .PHONY: start
 ifeq ($(RUN_MODE),local)
-start: install executor-image controller cli $(GOPATH)/bin/goreman
+start: install controller cli $(GOPATH)/bin/goreman
```
Comment (alexec) [inline Makefile:463]: this change means that we don't wait for the executor to be built before starting up, useful for UI only work

File: Procfile [hunk 2]
```diff
@@ -1,5 +1,6 @@
-controller: PNS_PRIVILEGED=true DEFAULT_REQUEUE_TIME=${DEFAULT_REQUEUE_TIME} LEADER_ELECTION_IDENTITY=local ALWAYS_OFFLOAD_NODE_STATUS=${ALWAYS_OFFLOAD_NODE_STATUS} OFFLOAD_NODE_STATUS_TTL=30s WORKFLOW_GC_PERIOD=30s UPPERIO_DB_DEBUG=${UPPERIO_DB_DEBUG} ARCHIVED_WORKFLOW_GC_PERIOD=30s ./dist/workflow-controller --executor-image ${IMAGE_NAMESPACE}/argoexec:${VERSION} --namespaced=${NAMESPACED} --namespace ${NAMESPACE} --loglevel ${LOG_LEVEL}
-argo-server: UPPERIO_DB_DEBUG=${UPPERIO_DB_DEBUG} ./dist/argo --loglevel ${LOG_LEVEL} server --namespaced=${NAMESPACED} --namespace ${NAMESPACE} --auth-mode ${AUTH_MODE} --secure=$SECURE --x-frame-options=SAMEORIGIN
-ui: yarn --cwd ui install && yarn --cwd ui start
+executor-image : make  executor-image
```
Comment (alexec) [inline Procfile:1]: we build it here instead

File: Makefile [hunk 3]
```diff
@@ -57,7 +55,9 @@ endif
 # * `local` run the workflow–controller and argo-server as single replicas on the local machine (default)
 # * `kubernetes` run the workflow-controller and argo-server on the Kubernetes cluster
 RUN_MODE              := local
-K3D                   := $(shell if [[ "`which kubectl`" != '' ]] && [[ "`kubectl config current-context`" == "k3d-"* ]]; then echo true; else echo false; fi)
+KUBECTX               := $(shell [[ "`which kubectl`" != '' ]] && kubectl config current-context || echo none)
```
Comment (alexec) [inline Makefile:58]: you only get `make nuke` for `KUBECTX=docker-desktop OR KUBECTX=k3d-*`

**Comment by alexec** [comment]
`Unable to connect to the server: dial tcp 10.1.0.36:6443: i/o timeout`

File: Makefile [hunk 4]
```diff
@@ -416,6 +420,18 @@ ifeq ($(RUN_MODE),kubernetes)
 	kubectl -n $(KUBE_NAMESPACE) scale deploy/argo-server --replicas 1
 endif
 
+# nuke is like "clean" but attempts to return you to a state as if it had never been installed at all
+# this only available in "safe" local development
+ifeq (true,$(filter true, $(DOCKER_DESKTOP) $(K3D)))
```
Comment (simster7) [inline Makefile:425]: `k3d` is not guaranteed to be a dev env, but if you're using `make nuke` you better know what you're doing.

File: Procfile (hunk 2 above)
Comment (simster7) [reply Procfile:1]: Do we have to wait to start workflows in this case? Is there a way to know when we are ready to do so? I guess Pods will just take longer to start up, although they could BackOff

**Review by simster7 — APPROVED** [review]

File: Makefile (hunk 4 above)
Comment (alexec) [reply Makefile:425]: I hope that "nuke" help people be careful - but k3d should only be desktop - k3s might not

**Comment by alexec** [comment]
> Do we have to wait to start workflows in this case? Is there a way to know when we are ready to do so? I guess Pods will just take longer to start up, although they could BackOff

Yes. This will always be shorter than current.

**Pull request merged**